### PR TITLE
add CONTRIBUTING.md with log level guidelines

### PR DESCRIPTION
**Pull request by ulzha**

### CODE REVIEW

**Comment by danielnorberg** [comment]
For reference, the k8s logging guidelines: https://github.com/kubernetes/community/blob/master/contributors/devel/logging.md

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+* `warn` when Styx cannot run a workflow or respond to an API call; e.g.
+  an external system being down. (Note: Styx design is presumed to be
+  resilient to intermittent failures in the first place, therefore an
+  external system being temporarily down is not a Styx error and should
+  not inevitably lead to one.)
+* `info` for other significant occurrences; e.g. starting and stopping,
+  responding to API calls.
+* `debug` for other occurrences, uncertain significance but eventually
+   helpful in troubleshooting; e.g. workflow instance state transitions.
+
```
Comment (danielnorberg) [inline CONTRIBUTING.md]: 👍

**Comment by honnix** [comment]
👍

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  by design. A ["This should never happen"](https://github.com/spotify/styx/blob/75f4463f52f964a131473cbc8597edd4d76834d3/styx-common/src/main/java/com/spotify/styx/util/Retrier.java#L74)
+  type of thing, symptomatic of a Styx bug. 
+* `warn` when Styx cannot run a workflow or respond to an API call; e.g.
+  an external system being down. (Note: Styx design is presumed to be
+  resilient to intermittent failures in the first place, therefore an
+  external system being temporarily down is not a Styx error and should
+  not inevitably lead to one.)
+* `info` for other significant occurrences; e.g. starting and stopping,
+  responding to API calls.
+* `debug` for other occurrences, uncertain significance but eventually
```
Comment (bergman) [inline CONTRIBUTING.md]: Is `eventually` the right word here? I would go for `possibly` or similar non-promising words.

File: README.md [hunk 3]
```diff
@@ -32,14 +32,15 @@ easily be added.
 
 ## Key concepts
 
-The key type of information that Styx concerns itself with are Workflows. A Workflow is either
+The key concept that Styx concerns itself with are Workflows. A Workflow is either
```
Comment (bergman) [inline README.md]: I think the ` … are Workflows` should be ` … is Workflows`.

**Comment by bergman** [comment]
Please reflow the paragraphs that you changed, some lines are too long and some too short.

**Pull request merged**